tests/shared/test_utils.py

```diff
@@ -18,6 +21,16 @@ def test_forward_fill():
     assert result == expected
 
 
+def test_forward_fill_fail():
+    # given:
+    x = [1, 2, 10, 4, None, None, 10, 10]
+
+    # when:
+    cond_value = None
+    with pytest.raises(OptimisationResultsError):
+        _ = utils.forward_fill(x, cond_value)
```

Comment:
@CharlyEmpereurmot can you tell me what are the cases where forward_fill should fail with an informative error? I though that an empty list would be one, but it is not the case.

Reply:
In fact we can probably check only that all results are not None, which means all simulations had been crashing because the setup of the optimization was incorrect. Also the IndexError handling here makes no sense and can be removed I would say.